skopt/plots.py

```diff
@@ -93,14 +93,17 @@ def plot_convergence(*args, **kwargs):
     return ax
 
 
-def _format_scatter_plot_axes(ax, space, ylabel):
+def _format_scatter_plot_axes(ax, space, ylabel, dim_labels=None):
     # Work out min, max of y axis for the diagonal so we can adjust
     # them all to the same value
     diagonal_ylim = (np.min([ax[i, i].get_ylim()[0]
                              for i in range(space.n_dims)]),
                      np.max([ax[i, i].get_ylim()[1]
                              for i in range(space.n_dims)]))
 
+    if dim_labels is None:
+        dim_labels = ["$X_{%i}$" % i for i in range(space.n_dims)]
```

Comment:
We could use `dimension.name` for those dimensions which have a name now that #508 has been merged. Default to X_i if there is no name. WDYT?